src/vcpkg/binarycaching.cpp

```diff
@@ -952,68 +952,71 @@ namespace
     struct GHABinaryProvider : IBinaryProvider
     {
         GHABinaryProvider(const VcpkgPaths& paths,
-                          bool read,
-                          bool write,
-                          const Optional<std::string>& url,
-                          const Optional<std::string>& token)
+                          const bool read,
+                          const bool write,
+                          const std::string& url,
+                          const std::string& token_header)
             : paths(paths)
         {
-            if (read) m_read_url = url.value_or_exit(VCPKG_LINE_INFO) + "_apis/artifactcache/cache";
-            if (write) m_write_url = url.value_or_exit(VCPKG_LINE_INFO) + "_apis/artifactcache/caches";
-            m_token_header = "Authorization: Bearer " + token.value_or_exit(VCPKG_LINE_INFO);
+            if (read) m_read_url = url;
+            if (write) m_write_url = url + "s";
+            m_token_header = token_header;
+            m_accept_header = "Accept: application/json;api-version=6.0-preview.1";
+            m_content_type_header = "Content-Type: application/json";
         }
 
-        Command command() const
+        std::string lookup_cache_entry(std::string& name, const std::string& abi) const
         {
-            Command cmd;
-            cmd.string_arg("curl")
-                .string_arg("-s")
-                .string_arg("-H")
-                .string_arg("Content-Type: application/json")
-                .string_arg("-H")
-                .string_arg(m_token_header)
-                .string_arg("-H")
-                .string_arg(m_accept_header);
-            return cmd;
-        }
+            auto url =
+                format_url_query(m_read_url, std::vector<std::string>{"keys=" + name + "-" + abi, "version=" + abi});
 
-        std::string lookup_cache_entry(const std::string& abi) const
-        {
-            auto cmd = command()
-                           .string_arg(m_read_url)
-                           .string_arg("-G")
-                           .string_arg("-d")
-                           .string_arg("keys=vcpkg")
-                           .string_arg("-d")
-                           .string_arg("version=" + abi);
+            auto res = invoke_http_request(
```

Comment:
We could also improve this by making only one curl call instead of one for every package. 
For example: `curl -w "8a1db05f-a65d-419b-aa72-037fb4d0672e %{http_code}\\n" https://httpbin.org/get https://httpbin.org/get`